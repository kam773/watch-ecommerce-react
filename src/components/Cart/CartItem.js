import React from 'react'

export default function CartItem({ item, value }) {
    const { id, title, img, price, total, count } = item;
    const { increment, decrement, removeItem } = value;
  return (
      <div className="row d-flex flex-row align-items-center my-2 text-capitalize text-center">
          <div className="col-10 mx-auto col-lg-2">
              <img
              src={ img }
              style={ { width: '5rem', height: 'auto' } }
              className="img-fluid"
              alt="product"/>
          </div>
          <div className="col-10 mx-auto col-lg-2">
              <span className="d-lg-none text-bold">product: </span>
              { title }
          </div>
          <div className="col-10 mx-auto col-lg-2">
              <span className="d-lg-none text-bold">price: </span>
              { price }
          </div>
          <div className="col-10 mx-auto col-lg-2 my-2 my-lg-0">
              <div className="d-flex justify-content-center">
                  <div>
                      <span className="btn btn-black mx-1 text-bold" onClick={ () => decrement(id) }>-</span>
                      <span className="btn btn-black mx-1 text-bold">{ count }</span>
                      <span className="btn btn-black mx-1 text-bold" onClick={ () => increment(id) }>+</span>
                  </div>
              </div>
          </div>
          {/*  */}
          <div className="col-10 mx-auto col-lg-2">
              <div className="cart-icon" onClick={ () => removeItem(id) }>
                  <i style={ { cursor: 'pointer' } } className="fas fa-trash"></i>
              </div>
          </div>
          <div className="col-10 mx-auto col-lg-2">
              <strong> item total : $ { total }</strong>
          </div>
      </div>
  )
}
