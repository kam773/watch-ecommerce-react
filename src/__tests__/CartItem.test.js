import React from 'react';
import CartItem from '../components/Cart/CartItem';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe("CartItem", () => {
    it("renders without crashes", () => {
        const props = {
            item: {},
            value: {}
        }
        const tree = renderer.create(<CartItem {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('has div with class', () => {
        const props = {
            item: {},
            value: {}
        }
      const wrapper = shallow(<CartItem {...props}/>)
      expect(wrapper.find('div').exists()).toEqual(true);
    });
    it('has 5 span elements', () => {
        const props = {
            item: {},
            value: {}
        }
      const wrapper = shallow(<CartItem {...props}/>)
      expect(wrapper.find('span')).toHaveLength(5);
    });
    it('has 1 strong element', () => {
        const props = {
            item: {},
            value: {}
        }
      const wrapper = shallow(<CartItem {...props}/>)
      expect(wrapper.find('strong')).toHaveLength(1);
    });
    it('has 1 img element', () => {
        const props = {
            item: {},
            value: {}
        }
      const wrapper = shallow(<CartItem {...props}/>)
      expect(wrapper.find('img')).toHaveLength(1);
    });
});
