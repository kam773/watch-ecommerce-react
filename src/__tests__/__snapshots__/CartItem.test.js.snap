// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders without crashes 1`] = `
<div
  className="row d-flex flex-row align-items-center my-2 text-capitalize text-center"
>
  <div
    className="col-10 mx-auto col-lg-2"
  >
    <img
      alt="product"
      className="img-fluid"
      style={
        Object {
          "height": "auto",
          "width": "5rem",
        }
      }
    />
  </div>
  <div
    className="col-10 mx-auto col-lg-2"
  >
    <span
      className="d-lg-none"
    >
      product: 
    </span>
  </div>
  <div
    className="col-10 mx-auto col-lg-2"
  >
    <span
      className="d-lg-none"
    >
      price: 
    </span>
  </div>
  <div
    className="col-10 mx-auto col-lg-2 my-2 my-lg-0"
  >
    <div
      className="d-flex justify-content-center"
    >
      <div
        className="sc-bdVaJa ksOhOh"
      >
        <span
          className="btn btn-black mx-1on>"
          onClick={[Function]}
        >
          -
        </span>
        <span
          className="btn btn-black mx-1on>"
        />
        <span
          className="btn btn-black mx-1on>"
          onClick={[Function]}
        >
          +
        </span>
      </div>
    </div>
  </div>
  <div
    className="col-10 mx-auto col-lg-2"
  >
    <div
      className="cart-icon"
      onClick={[Function]}
    >
      <i
        className="fas fa-trash"
        style={
          Object {
            "cursor": "pointer",
          }
        }
      />
    </div>
  </div>
  <div
    className="col-10 mx-auto col-lg-2"
  >
    <strong>
       item total : $ 
    </strong>
  </div>
</div>
`;
